src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -1181,11 +1185,18 @@ UniqueAVFrame SingleStreamDecoder::decodeAVFrame(
       decodeStats_.numPacketsRead++;
 
       if (status == AVERROR_EOF) {
-        // End of file reached. We must drain the codec by sending a nullptr
-        // packet.
-        status = avcodec_send_packet(
-            streamInfo.codecContext.get(),
-            /*avpkt=*/nullptr);
+        // End of file reached. We must drain the decoder
+        if (useCustomInterface) {
+          AutoAVPacket eofAutoPacket;
+          ReferenceAVPacket eofPacket(eofAutoPacket);
+          eofPacket->data = nullptr;
+          eofPacket->size = 0;
```

Comment:
We should be able to define a global, constant EOF packet that is local to this cpp file. Or if we can't do that (because maybe we need to imperatively set the fields `data` and `size` and we can't do it on object construction?) we can define a `getEoFPacket()` function which does this.

Reply:
I'll add a P0 TODO to address this. Defining a global isn't possible as you mentioned, and having `getEoFPacket()` isn't easy either because of the relationship between AutoAVPacket and ReferenceAVPacket. I think we'll be able to clean that all up by adding a new `sendEOFPacket()` to the interface, eventually.